pkg/model/core/resource.go

```diff
@@ -44,3 +44,11 @@ func (m *ResourceMeta) ID() string {
 type ResourceVisitor interface {
 	Visit(res Resource) error
 }
+
+type EventType string
+
+const (
+	CreateEvent EventType = "Create"
+	UpdateEvent EventType = "Update"
+	DeleteEvent EventType = "Delete"
+)
```

Comment:
controller also defines event types, why we need own? https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.2/pkg/event

Reply:
The controller runtime event definitions are structs, and includes a GenericEvent. In order to use them, we'd have to create an interface for them and that would be the new EventType. Ultimately, it's more work with no benefit compared to using consts.